src/main/java/org/candlepin/subscriptions/db/model/Host.java

```diff
@@ -277,7 +286,9 @@ public HostTallyBucket addBucket(String productId, ServiceLevel sla, Usage usage
 
     public void addBucket(HostTallyBucket bucket) {
         bucket.getKey().setHost(this);
-        getBuckets().add(bucket);
+        if (!getBuckets().contains(bucket)) {
```

Comment:
Why not make this a Set<HostTallyBucket> instead?

Reply:
I dig it! Will try that. Commit incoming.